### PR TITLE
Yairp/test/workflow runner tests

### PR DESCRIPTION
**Pull request by yairper**
### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

File: packages/workflow-core/src/lib/workflow-runner.test.ts [hunk 1]
```diff
@@ -1,34 +1,501 @@
...
-    extensions: {
-      statePlugins: [],
+    final: {
+      on: { EVENT: 'initial' },
     },
+  },
+};
+
+function EventCollectingWorkflow(args) {
+  var wf = new WorkflowRunner(args);
```
Comment (alonp99) [inline packages/workflow-core/src/lib/workflow-runner.test.ts:28]: change to const 🙏

File: packages/workflow-core/src/lib/workflow-runner.test.ts [hunk 2]
```diff
@@ -1,34 +1,340 @@
...
-    extensions: {
-      statePlugins: [],
+    final: {
+      on: { EVENT: 'initial' },
     },
+  },
+};
+
+function EventCollectingWorkflow(args) {
+  const wf = new WorkflowRunner(args);
```
Comment (Omri-Levy) [inline packages/workflow-core/src/lib/workflow-runner.test.ts:28]: I'd prefer we avoid names with less than three characters.

File: packages/workflow-core/src/lib/workflow-runner.test.ts [hunk 3]
```diff
@@ -1,34 +1,340 @@
...
+  wf.events = [];
+  wf.subscribe(e => {
+    e.error && (e.error = e.error.message);
+    wf.events.push(e);
+  });
+  return wf;
+}
+
+const sleep = ms => new Promise(resolve => setTimeout(() => resolve(), ms));
```
Comment (Omri-Levy) [inline packages/workflow-core/src/lib/workflow-runner.test.ts:37]: If `sleep` ends up being in use let's use the one exported from `@ballerine/common`.

File: packages/workflow-core/src/lib/workflow-runner.test.ts [hunk 4]
```diff
@@ -1,34 +1,340 @@
...
     },
-    extensions: {
-      statePlugins: [],
+    final: {
+      on: { EVENT: 'initial' },
     },
+  },
+};
+
+function EventCollectingWorkflow(args) {
```
Comment (Omri-Levy) [inline packages/workflow-core/src/lib/workflow-runner.test.ts:27]: Thoughts about renaming it to something like `createEventCollectingWorkflow`? Would make it clearer that the function returns something and more or less what it does. The current capitalization might make it easy to confuse with a class.

File: sdks/workflow-browser-sdk/src/lib/workflow-browser-sdk.ts [hunk 5]
```diff
@@ -229,7 +229,7 @@ export class WorkflowBrowserSDK {
   }
 
   sendEvent(event: WorkflowEventWithoutState) {
```
Comment (Omri-Levy) [inline sdks/workflow-browser-sdk/src/lib/workflow-browser-sdk.ts:231]: Let's add the `async` keyword here as well now that `sendEvent` is async in `core`.

File: packages/workflow-core/src/lib/workflow-runner.test.ts [hunk 6]
```diff
@@ -1,34 +1,340 @@
...
+
+    const events = [];
+    wf.subscribe(event => events.push(event));
+    wf.subscribe(event => {});
+    wf.sendEvent({ type: 'EVENT', ...DEFAULT_PAYLOAD });
+
+    expect(events).toStrictEqual([]);
+  });
+
+  describe('transition plugins', () => {
```
[the remark this answers is not in the document]
Comment (yairper) [reply packages/workflow-core/src/lib/workflow-runner.test.ts:130]: talked about it, leaving as is...

File: sdks/workflow-browser-sdk/src/lib/workflow-browser-sdk.ts (hunk 5 above)
Comment (yairper) [reply sdks/workflow-browser-sdk/src/lib/workflow-browser-sdk.ts:231]: fixed, also made the same change in `workflow-node-sdk.ts`

File: packages/workflow-core/src/lib/workflow-runner.test.ts [hunk 7]
```diff
@@ -1,34 +1,340 @@
...
+                action: () => {
+                  throw new Error();
+                },
+              },
+            ],
+          },
+        });
+
+        await wf.sendEvent({ type: 'EVENT', ...DEFAULT_PAYLOAD });
+        await sleep(1);
```
Comment (alonp99) [inline packages/workflow-core/src/lib/workflow-runner.test.ts:181]: Why do we need the `sleep`s?

**Review by alonp99 — APPROVED** [review]

File: packages/workflow-core/src/lib/workflow-runner.test.ts (hunk 7 above)
Comment (alonp99) [reply packages/workflow-core/src/lib/workflow-runner.test.ts:181]: Now I can see that `__handleAction` is async and not being awaited, might be the source of then issue

**Pull request merged**